process/smartContract/process_test.go

```diff
@@ -2798,14 +2802,29 @@ func TestScProcessor_CreateCrossShardTransactionsWithAsyncCalls(t *testing.T) {
 	require.Equal(t, len(outputAccounts), len(scTxs))
 	require.True(t, createdAsyncSCR)
 
-	lastScTx := scTxs[len(scTxs)-1].(*smartContractResult.SmartContractResult)
-	require.Equal(t, vmData.AsynchronousCallBack, lastScTx.CallType)
-	require.Equal(t, lastScTx.Data, []byte("@"+core.ConvertToEvenHex(int(vmcommon.Ok))))
+	t.Run("backwards compatibility", func(t *testing.T) {
+		lastScTx := scTxs[len(scTxs)-1].(*smartContractResult.SmartContractResult)
+		require.Equal(t, vmData.AsynchronousCallBack, lastScTx.CallType)
+		require.Equal(t, []byte(nil), lastScTx.Data)
+	})
+	arguments.EpochNotifier.CheckEpoch(
+		&block.Header{
+			Epoch: 1,
+		},
+	)
+
+	createdAsyncSCR, scTxs, err = sc.processSCOutputAccounts(&vmcommon.VMOutput{GasRemaining: 1000}, vmData.AsynchronousCall, outputAccounts, tx, txHash)
+	t.Run("fix enabled, data field is correctly populated", func(t *testing.T) {
+		lastScTx := scTxs[len(scTxs)-1].(*smartContractResult.SmartContractResult)
+		require.Equal(t, vmData.AsynchronousCallBack, lastScTx.CallType)
+		require.Equal(t, []byte("@"+core.ConvertToEvenHex(int(vmcommon.Ok))), lastScTx.Data)
+	})
 
 	tx.Value = big.NewInt(0)
 	scTxs, err = sc.processVMOutput(&vmcommon.VMOutput{GasRemaining: 1000}, txHash, tx, vmData.AsynchronousCall, 10000)
 	require.Nil(t, err)
 	require.Equal(t, 1, len(scTxs))
+	lastScTx := scTxs[len(scTxs)-1].(*smartContractResult.SmartContractResult)
```

Comment:
useless line? the same value if overwritten in the next line

Reply:
refactor